data/templates/firewall/nftables-nat.j2

```diff
@@ -43,4 +56,8 @@ table ip vyos_nat {
 
 {{ group_tmpl.groups(firewall_group, False, True) }}
 }
+
+{%     if ip_fqdn.items() %}
+# use-domain-resolver=yes
```

Comment:
It does nothing
Is it correct?

Reply:
It is used a flag, in oder to start/stop vyos-domain resolver.
- https://github.com/vyos/vyos-1x/pull/4024/files#diff-73b0ee812456b3cb072d98d47135c69929642c2a72b52712fb1c130f14439732R261
- https://github.com/vyos/vyos-1x/pull/4024/files#diff-22639aad95616c8d057fd5f8e672c148e7d29e26077361e9175e089971c0f267R579

Reply:
@nicolas-fort The question is why the line is always generated with a `# ` before it — that means the line is commented out and will not have any effect, unless Viacheslav and I are missing anything about the config file syntax.

Reply:
Yes, to nftables it does nothing. As you said, it's always generated with `#`, so it's a comment. But while applying firewall and nat rules, it checks if that comment is present in those files, in order to start/stop vyos-domain-resolver service.

Reply:
@dmbaturin The line seems to be picked up by a file contents check in the conf script.

Though not sure why the `ip_fqdn` boolean check is insufficient here @nicolas-fort ?

Reply:
`vyos-domain-resolver` now updates domains and/or fqdn entries, if they are configured in firewall and NAT.
1. While parsing firewall rules, if fqdn or group-domains are used, then this comment is added to /run/nftables.conf as a flag that will be checked by nat scripts (of course, if nat is configured)
2. While parsing nat rules, if fqdn are used, then this comment is added to /run/nftables_nat.conf as a flag that will be checked by firewall scripts (of course, if firewall is configured).

Since only once service handles this translations, while parsing one part of the config (for example firewall), we need to know if fqdn are configured in the other feature (nat).

I thought that would be better this approach, and check if comments are present in those files in order to decide to stop/start vypos-domain-resolver service, rather than while parsing one feature (for example nat), also grab the full configuration of the other feature (in the example would be firewall), and then decide if the service should be started or stopped.

Reply:
I have a bad feeling about such action at a distance, to be fair. At the very least it should be documented in the generated file, why a line that has no effect within that file is included in it.

But I think we should look for other solutions before we take this route. In the worst case, can we just enable the resolver daemon by default? It does nothing when there are no domain groups, and doesn't take much memory or CPU when idle, does it?